pysph/sph/surface_tension.py

```diff
@@ -500,72 +820,225 @@ def loop(self, d_idx, s_idx, d_V, s_V, d_rho, s_rho,
+                dest=i, sources=fluids+solids, pb=0.0))
+            equations.append(MomentumEquationViscosityAdami(dest=i,
+                                                            sources=fluids))
+            equations.append(CSFSurfaceTensionForceAdami(dest=i, sources=None))
+            if len(solids) != 0:
+                equations.append(SolidWallNoSlipBC(dest=i, sources=solids,
+                                 nu=nu))
+        Equations.append(Group(equations))
+    elif scheme == 'shadloo':
+        Equations = []
+        equations = []
+        for i in fluids+solids:
+            equations.append(SummationDensity(dest=i, sources=fluids+solids))
+        Equations.append(Group(equations, real=real))
+        equations = []
+        for i in fluids:
+            equations.append(StateEquation(dest=i, sources=None, rho0=rho0,
+                                           p0=p0, b=b))
+            equations.append(SY11ColorGradient(dest=i, sources=fluids+solids))
+        for i in solids:
+            equations.append(SolidWallPressureBCnoDensity(dest=i,
+                                                          sources=fluids))
+        Equations.append(Group(equations, real=real))
+        equations = []
+        for i in fluids:
+            equations.append(ScaleSmoothingLength(dest=i, sources=None,
+                                                  factor=factor1))
+        Equations.append(Group(equations, real=real, update_nnps=True))
+        equations = []
+        for i in fluids:
+            equations.append(SY11DiracDelta(dest=i, sources=fluids+solids))
+        Equations.append(Group(equations, real=real))
+        equations = []
+        for i in fluids:
+            equations.append(InterfaceCurvatureFromNumberDensity(
+                dest=i, sources=fluids+solids, with_morris_correction=True))
+        Equations.append(Group(equations, real=real))
+        equations = []
+        for i in fluids:
+            equations.append(ScaleSmoothingLength(dest=i, sources=None,
+                                                  factor=factor2))
+        Equations.append(Group(equations, real=real, update_nnps=True))
+        equations = []
+        for i in fluids:
+            equations.append(MomentumEquationPressureGradient(
+                dest=i, sources=fluids+solids, pb=0.0))
+            equations.append(MomentumEquationViscosity(dest=i, sources=fluids,
+                                                       nu=nu))
+            equations.append(ShadlooYildizSurfaceTensionForce(dest=i,
+                                                              sources=None,
+                                                              sigma=sigma))
+            if len(solids) != 0:
+                equations.append(SolidWallNoSlipBC(dest=i, sources=solids,
+                                                   nu=nu))
+        Equations.append(Group(equations))
+    else:
```

Comment:
What scheme is the else case for?

Reply:
The else is for Morris Scheme. The function returns the Morris scheme by default if nothing is chosen.